jetbrains-core/tst/software/aws/toolkits/jetbrains/core/AwsClientManagerTest.kt

```diff
@@ -173,6 +183,29 @@ class AwsClientManagerTest {
         assertThat(second).isSameAs(first)
     }
 
+    @Test
+    fun userAgentIsPassed() {
+        val wireMockServer = WireMockServer(WireMockConfiguration.wireMockConfig())
+        try {
+            wireMockServer.start()
+
+            stubFor(any(anyUrl()).willReturn(aResponse().withStatus(200)))
+
+            val sut = getClientManager().createNewClient(
+                LambdaClient::class,
```

Comment:
probably overkill to loop through all the possible clients?

Reply:
ya, all logic is same once we have the builder method (which is reflection)